testnet/stacks-node/src/monitoring.rs

```diff
@@ -0,0 +1,51 @@
+use async_std::net::{TcpListener, TcpStream};
+use async_std::prelude::*;
+use async_std::task;
+use stacks::prometheus::{Encoder, TextEncoder, gather};
+
+pub use stacks::monitoring::{ERRORS_EMITTED_COUNTER, WARNING_EMITTED_COUNTER};
+
+use http_types::{
+    Response, 
+    StatusCode, 
+    Body,
+};
+
+pub fn start_serving_prometheus_metrics(prometheus_bind: String) {
+    let addr = prometheus_bind.clone();
+
+    async_std::task::block_on(async {
+        let listener = TcpListener::bind(addr).await.expect("todo(ludo)");
+        let addr = format!("http://{}", listener.local_addr().expect("todo(ludo)"));
+        println!("Prometheus server listening on {}", addr);
+    
+        let mut incoming = listener.incoming();
+        while let Some(stream) = incoming.next().await {
+            let stream = stream.expect("todo(ludo)");
+            let addr = addr.clone();
+    
+            task::spawn(async {
+                if let Err(err) = accept(addr, stream).await {
+                    eprintln!("{}", err);
+                }
+            });
+        }      
+    });
+}
+
+async fn accept(addr: String, stream: TcpStream) -> http_types::Result<()> {
+    println!("starting new connection from {}", stream.peer_addr()?);
+    async_h1::accept(&addr, stream.clone(), |_| async {
+        let encoder = TextEncoder::new();    
```

Comment:
I'm glad to see that Prometheus separates the task of gathering monitoring data from the task of sending it over the network.  This means we can easily add a HTTP endpoint for Prometheus statistics in the near future (e.g. `/v2/monitoring/prometheus` or something), and strip this module out completely.

Reply:
Sure, however we talked as a team about the different possible approaches (and I presented the one you're suggesting) before working on this, and went with the consensus.

Reply:
Should we revisit now?

Reply:
I could see reasons for and against doing this -- on one hand, if this is really a compiler-time option, keeping the module separate from the HTTP code will make it a lot cleaner of a separation, on the other hand, integrating it reduces the number of HTTP servers. Though having metrics collection share the same thread as real requests is generally frowned upon.

Reply:
We don't have to do it right now, but we should do it at some point before mainnet, once we're sure about the endpoints.  We can (and should) have a separate thread for serving monitoring data.